### ISSUE TITLE
Client side properties: "Invalid signature" when writing a uint32 property

### ISSUE THREAD

**Issue by mvidner** [mvidner/ruby-dbus#108]
Properties defined as `uint32` are being sent as `int32`.  In this specific case, bluetooth discovery timeout, as defined [here](https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/doc/adapter-api.txt).

```ruby

adapter[ 'DiscoverableTimeout' ] = 0
```

```
/usr/local/lib/ruby/gems/2.7.0/gems/ruby-dbus-0.18.0.beta2/lib/dbus/bus.rb:368:in `block in send_sync_or_async': Invalid signature for 'DiscoverableTimeout'; caused by 3 sender=:1.7 -> dest=:1.35 serial=320 reply_serial=20 path=; interface=; member= error_name=org.freedesktop.DBus.Error.InvalidSignature (DBus::Error)
```
... and on the `dbus-monitor` end of things:
```
method call time=1649120833.192372 sender=:1.36 -> destination=org.bluez serial=20 path=/org/bluez/hci0; interface=org.freedesktop.DBus.Properties; member=Set
   string "org.bluez.Adapter1"
   string "DiscoverableTimeout"
   variant       int32 0
error time=1649120833.192665 sender=:1.7 -> destination=:1.36 error_name=org.freedesktop.DBus.Error.InvalidSignature reply_serial=20
   string "Invalid signature for 'DiscoverableTimeout'"
```

This is using the beta2 prerelease.

_Originally posted by @mahlonsmith in https://github.com/mvidner/ruby-dbus/issues/97#issuecomment-1088170282_

**Comment by mvidner** [comment]
Status:

I thought this would be a simple thing to include in today's beta5 release. In 97 I said:
> It seems the fix will be using the recently introduced Data.make_typed in ProxyObjectInterface#[]=.

which was correct and looked easy, but of course then we need the correct *type* argument. For that, ProxyObjectInterface and a bunch of related classes need to process the introspection data for properties, not only methods and signals. Straightforward but not trivial, postponing for later this week.

**Comment by mvidner** [comment]
Fixed in #113, released in https://rubygems.org/gems/ruby-dbus/versions/0.18.0.beta7
